File: src/main/java/me/testaccount666/serversystem/clickablesigns/SignClickAction.java
package me.testaccount666.serversystem.clickablesigns;

import me.testaccount666.serversystem.userdata.User;
import org.bukkit.block.Sign;

public interface SignClickAction {
    void execute(User user, Sign sign);

    /**
     * The permission node for the permission required to use this sign.
     *
     * @return The permission node
     */
    String getUsePermissionNode();

    /**
     * The permission node for the permission required to break this+ sign.
     *
     * @return The permission node
     */
    String getDestroyPermissionNode();

    String getBasePermissionNode();
}

File: src/main/kotlin/me/testaccount666/serversystem/clickablesigns/SignClickAction.kt
package me.testaccount666.serversystem.clickablesigns

import me.testaccount666.serversystem.userdata.User
import org.bukkit.block.Sign

interface SignClickAction {
    fun execute(user: User, sign: Sign)

    /**
     * The permission node for the permission required to use this sign.
     * 
     * @return The permission node
     */
    val usePermissionNode: String

    /**
     * The permission node for the permission required to break this+ sign.
     * 
     * @return The permission node
     */
    val destroyPermissionNode: String

    val basePermissionNode: String
}
